add example for controlled components (#6519)
redwoodjs/redwood: docs/docs/forms.md

}
```

### Controlled Component Fields

If you're working with a fully-featured component library, or have your own production-ready components, you may want to integrate them with Redwood's forms seamlessly.
You can via Redwood forms' `useErrorStyles` hook and React Hook Form's `Controller` component.
The following example shows how you could go about integrating a component from [`primereact`](https://www.primefaces.org/primereact/) for use in in Redwood's forms like any of the named-input fields listed above:

> ```jsx title="web/src/components/ToggleButtonField/ToggleButtonField.tsx"

Reviewer: ```suggestion
```tsx title="web/src/components/ToggleButtonField/ToggleButtonField.tsx"
```

}
```

### Controlled Component Fields

> If you're working with a fully-loaded component library, or bring your own production-ready components you may want them to integrate with Redwood seemlessly.
> 
> You can create these custom fields through the use of Redwood's `useErrorStyles` and React Hook Form's `Controller`. 
> 
> The following example shows a component from [`primereact`](https://www.primefaces.org/primereact/) which can be used in Forms like any of the named input fields (listed above).

Reviewer: ```suggestion
If you're working with a fully-featured component library, or have your own production-ready components, you may want to integrate them with Redwood's forms seamlessly.
You can via Redwood forms' `useErrorStyles` hook and React Hook Form's `Controller` component.
The following example shows how you could go about integrating a component from [PrimeReact](https://www.primefaces.org/primereact/) for use in in Redwood's forms like any of the named-input fields listed above:
```

> ```jsx

Reviewer: Could you change this to: ```tsx title="web/src/components/ToggleButtonField.tsx"

GitHub's suggestion feature can't handle the backticks

> import { ToggleButton, ToggleButtonProps } from 'primereact/togglebutton'

Reviewer: Just a personal preference:
```suggestion
import { ToggleButton } from 'primereact/togglebutton'
import type { ToggleButtonProps } from 'primereact/togglebutton'
```

Reviewer: Another personal preference, just helps upfront for readability to say it's a type

import { Controller, RegisterOptions, useErrorStyles } from '@redwoodjs/forms'

> interface Prop extends ToggleButtonProps {

Reviewer: ```suggestion
interface Props extends ToggleButtonProps {
```

  validation?: RegisterOptions
  errorClassName?: string
}

> const ToggleButtonField = (prop: Prop) => {

Reviewer: ```suggestion
const ToggleButtonField = (props: Props) => {
```

  const {
    name,
    className,
    errorClassName,
    defaultValue,
    validation,
    style,

>     ...propRest
>   } = prop

Reviewer: ```suggestion
    ...propsRest
  } = props
```

Reviewer: Personal preference here, I've just always seen it pluralized

Author: I usually argue that the pluralized form should be used if the arguments contain multiple objects to clarify that the props can be destructed multiple times. This is definitely the case here. Thanks for your feedback.

  const { className: componentClassName, style: componentStyle } =
    useErrorStyles({
      className: className,
      errorClassName: errorClassName,
      name: name,
    })

  return (
    <Controller
      name={name}
      defaultValue={defaultValue}
      rules={validation}
      render={({ field: { onChange, onBlur, value, name, ref } }) => (
        <ToggleButton

>           {...propRest}

Reviewer: ```suggestion
          {...propsRest}
```

          checked={value}
          onChange={onChange}
          onBlur={onBlur}
          ref={ref}
          name={name}
          className={componentClassName}
          style={{ ...componentStyle, ...style }}
        />
      )}
    />
  )
}

export default ToggleButtonField

> 

Reviewer: ```suggestion
```

Reviewer: (just removing the empty space at the bottom of the code block)
